src/carbonio-files-ui-common/views/components/SelectionProvider.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { createContext, useCallback, useContext, useEffect, useMemo, useState } from 'react';
+
+import { useReactiveVar } from '@apollo/client';
+import { isEqual } from 'lodash';
+
+import { selectionModeVar } from '../../apollo/selectionVar';
+import { useMemoCompare } from '../../hooks/useMemoCompare';
+
+function filterSelectedIDs(selectedIDs: string[], items: { id: string }[]): string[] {
+	return selectedIDs.filter((selectedId) => items.some((item) => item.id === selectedId));
+}
+
+interface SelectionContextType {
+	selectedIDs: string[];
+	selectedMap: { [id: string]: boolean };
+	selectId: (id: string) => void;
+	isSelectionModeActive: boolean;
+	unSelectAll: () => void;
+	selectAll: () => void;
+	exitSelectionMode: () => void;
+	selectedCount: number;
+	isAllSelected: boolean;
+}
+
+export function useSelection(items: Array<{ id: string }>): SelectionContextType {
+	const [selectedIDs, setSelectedIDs] = useState<string[]>([]);
+	const [isSelectionModeActive, setIsSelectionModeActive] = useState(false);
+	const selectionModeActive = useReactiveVar(selectionModeVar);
```

Comment:
Isn't the state redundant if we have the reactive var? The current usage always updates the state with the same value of the reactive var. We could use the reactive var directly, couldn't we?